app/models/school_data_importer.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require "file_download"
+require "csv"
+
+class SchoolDataImporter
+  # TODO: Register and Partner 147: Figure out how exactly we want to handle school updating
+  def run
+    CSV.foreach(schools_data_file.path, headers: true, encoding: "ISO-8859-1:UTF-8") do |row|
+      school = row_to_school(row)
+      school.save!
+    end
+  end
+
+private
+
+  def gias_schools_csv_url
+    date_string = Time.zone.now.strftime("%Y%m%d")
+    "https://ea-edubase-api-prod.azurewebsites.net/edubase/downloads/public/edubasealldata#{date_string}.csv"
+  end
+
+  def schools_data_file
+    FileDownload.new(gias_schools_csv_url).fetch
+  end
+
+  def row_to_school(row)
+    school = School.find_or_initialize_by(urn: row.fetch("URN"))
```

Comment:
School.find_or_initialize_by(urn: row.fetch("URN")) do |school|
   school.name = row.fetch("EstablishmentName") .... would save you having to have school as the last statement just so we can return. Even more I doubt we want to be doing these assignments for schools that have already been created.

Reply:
I think we should make those assignments for schools we already have in database - cause they can change, and that's an easy way to update them.